standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -1535,20 +1556,14 @@ public void setHadoopJobId(String hadoopJobId, long id) {
   @Override
   @RetrySemantics.Idempotent
   public long findMinOpenTxnIdForCleaner() throws MetaException {
-    Connection dbConn = null;
     try {
-      try {
-        dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED, connPoolCompaction);
+      try (Connection dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED, connPoolCompaction)) {
         return getMinOpenTxnIdWaterMark(dbConn);
       } catch (SQLException e) {
-        LOG.error("Unable to getMinOpenTxnIdForCleaner", e);
-        rollbackDBConn(dbConn);
-        checkRetryable(e, "getMinOpenTxnForCleaner");
-        throw new MetaException("Unable to execute getMinOpenTxnIfForCleaner() " +
-            e.getMessage());
-      } finally {
-        closeDbConn(dbConn);
-      }
+        LOG.error("Unable to findMinOpenTxnIdForCleaner", e);
+        checkRetryable(e, "findMinOpenTxnIdForCleaner");
+        throw new MetaException("Unable to execute getMinOpenTxnIfForCleaner() " + e.getMessage());
```

Comment:
nit: Typo - `getMinOpenTxnIdForCleaner` instead of `getMinOpenTxnIfForCleaner`

Reply:
fixed